src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/LicenceConditionsCoordinator.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.client.DeliusClient.LicenceConditions
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.LicenceConditionResponse
+import java.util.Collections
+
+@Service
+class LicenceConditionsCoordinator {
+  internal fun selectLicenceConditions(nDeliusLicenceConditions: LicenceConditions, cvlLicenceConditions: List<LicenceConditionResponse?>): SelectedLicenceConditions {
+    val hasAllConvictionsReleasedOnLicence = hasAllConvictionsReleasedOnLicence(nDeliusLicenceConditions)
+    val onLicenceCvlWithLaterOrSameStartDate = isCvlLicenceMoreOrAsRecent(nDeliusLicenceConditions, cvlLicenceConditions, hasAllConvictionsReleasedOnLicence == true)
+    val source = if (onLicenceCvlWithLaterOrSameStartDate) "cvl" else "nDelius"
+
+    return SelectedLicenceConditions(
+      source = source,
+      hasAllConvictionsReleasedOnLicence = hasAllConvictionsReleasedOnLicence,
+      ndeliusLicenceConditions = nDeliusLicenceConditions,
+      cvlLicenceConditions = cvlLicenceConditions
+    )
+  }
+
+  private fun hasAllConvictionsReleasedOnLicence(nDeliusLicenceConditions: LicenceConditions): Boolean? {
+    return if (nDeliusLicenceConditions.activeConvictions.isNotEmpty()) {
+      nDeliusLicenceConditions.activeConvictions.all {
+        it.sentence?.isCustodial == true && it.sentence.custodialStatusCode == "B"
```

Comment:
I don't think there's a test that explains/tests what this line of code is doing and why it's needed.

Reply:
> I don't think there's a test that explains/tests what this line of code is doing and why it's needed.

custodialStatusCode 'B' means 'on licence'. 
This is tested at the controller level and in the unit tests. For example, if you change the value to 'A' multiple tests will fail. 
The LicenceConditionsControllerTest has a function 
` licenceConditionsResponse(crn = crn, releasedOnLicence = true ....)`
that sets this value in it's Wiremock stubs (actually MockServer) `releasedOnLicence = true` sets the value to 'B'

Reply:
Done

Reply:
By the way, I've noticed that this PR is almost a whopping 900 lines long!!
Do you think it would be better to make my PRs shorter in future for readability?
For example I could make two <500 line PRs instead of one 1000 line PR.

Reply:
I think yes, shorter PRs are better but sometimes it's just not practical.  I did get to the point with this one where I thought "that's enough, I'm going to skip the remainder", which isn't a good thing.

Reply:
The point about the "B" in the tests has 2 sides to it:
1) What you said about B meaning "On Licence" is the bit of info that should be very apparent in the tests.  I don't see anywhere that explains it.
2) The tests don't require the presence of the fragment of code `&& it.sentence.custodialStatusCode == "B"` 
If you comment that out, the tests still pass.  That's a sign that the tests could be improved.

Reply:
> 1. What you said about B meaning "On Licence" is the bit of info that should be very apparent in the tests. I don't see anywhere that explains it.

It's there but not obvious. I've rewritten the tests in a way to hopefully make it stand out more
![image](https://github.com/ministryofjustice/make-recall-decision-api/assets/104494371/efbb93d2-45ca-474b-a3bb-c1b4945d970f)

>2.  The tests don't require the presence of the fragment of code && it.sentence.custodialStatusCode == "B"
> If you comment that out, the tests still pass. That's a sign that the tests could be improved.
The tests do actually fail you remove that fragment. 
![image](https://github.com/ministryofjustice/make-recall-decision-api/assets/104494371/90d4669d-3a35-4fb3-b0a1-7a13857390b3)